src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -159,14 +216,35 @@ public EditPropertyDescriptor(EditPropertyDescriptor toCopy) {
             setSellerEmail(toCopy.sellerEmail);
             setPrice(toCopy.price);
             setTags(toCopy.tags);
+            setTagsToAdd(toCopy.tagsToAdd);
+            setTagsToDelete(toCopy.tagsToDelete);
         }
 
         /**
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
             return CollectionUtil.isAnyNonNull(propertyName, sellerPhone, sellerEmail,
-                    address, tags, sellerName, price);
+                    address, tags, tagsToAdd, tagsToDelete, sellerName, price);
+        }
+
+        /**
+         * Returns true if tags are reset and tags are modified by addition or deletion.
+         */
+        public boolean isTagsBothResetAndModified() {
+            return CollectionUtil.isAnyNonNull(tags) && (CollectionUtil.isAnyNonNull(tagsToAdd, tagsToDelete));
```

Comment:
```suggestion
            return tags!=null && (CollectionUtil.isAnyNonNull(tagsToAdd, tagsToDelete));
```
Suggestion for clarity that it is not checking if the tags contains null